### ISSUE TITLE
No Sound on HomePod Mini

### ISSUE THREAD

**Issue by niteguide** [music-assistant/support#3131]
### What version of Music Assistant has the issue?

2.3.2

### What version of the Home Assistant Integration have you got installed?

2024.10.4

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

When I start a playlist after some given time sound output stops on HomePod Mini. Song gets stuck (say for example a song that is 3:00 minutes long is shown as playing for 15:00 minutes ... ). No such issues with Google Cast players. 
Reason why I would like to bring this up again after #2863 had been closed: I am very sure that HomePod Mini worked with my network setup already with some previous versions of MA. 
I have reset HomePod Mini and router and disabled firewall. 

### How to reproduce

Start a playlist. At some point sound output stops on HomePod Mini. Music Assistant shows: 
![IMG_2599](https://github.com/user-attachments/assets/21c932fb-fce7-43b9-9c6e-4ea8fd13eb71)

### Music Providers

Happens regardless of music provider (tried with Tidal, Spotify). 

### Player Providers

HomePod Mini/Airplay 

### Full log output

This is a part of a debug log. It starts with the song that got stuck (Uxxii - Autumn Visions); Chromecast continues to play. 
[debug_part.docx](https://github.com/user-attachments/files/17622670/debug_part.docx)

### What version of Home Assistant Core are your running

2014.10.4

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by OzGav** [comment]
I see someone else remarking that having a Sonos player added to HomeKit caused a problem. Is your Homepod connected to any other service?

**Comment by niteguide** [comment]
> I see someone else remarking that having a Sonos player added to HomeKit caused a problem. Is your Homepod connected to any other service?

the HomePod is indeed added to HomeKit and also in use as a home hub for HomeKit devices, in fact I was not aware that you can use it without having it in the Home-App ...

**Comment by OzGav** [comment]
I would try removing is from everything but MA then and see if that changes anything.

**Comment by niteguide** [comment]
> I would try removing is from everything but MA then and see if that changes anything.

unfortunately this seems not possibe; the HomePod does loose all network settings (Wifi) as soon as I delete it from the Home-App;

**Comment by OzGav** [comment]
I don't use these so I am potentially suggesting something not possible but is Homekit a different thing?

**Comment by niteguide** [comment]
> I don't use these so I am potentially suggesting something not possible but is Homekit a different thing?

I did some research - generally it is not possible to setup and operate HomePod Mini without adding it to HomeKit. When you don't have an iPhone you will need some third party app to get it work - so using it outside of HomeKit is not intended by Apple - when I am not missing something; 
With Sonos this is of course different as these speakers have a dedicated app and are not dependent on HomeKit;

**Comment by OzGav** [comment]
Ok scratch that theory then…

**Comment by JonathanKang** [comment]
I have the same issue with my ayino wr320. And I can re-produce this issue playing specific songs with airplay and linkplay player entity, but not dlna.

When playing such songs with airplay entities, the behaviour is the same as what is described in this report. While playing such songs with linkplay speaker entity, the music pauses instead.

**Comment by niteguide** [comment]
I have now tested the HomePod Mini within a completely different network (other router brand, other provider, different location, other raspberry) - after a few attempts I get the exact same error.
I am still however looking for a pattern. Sometimes HomePod Mini plays 8+ hours without any problems and I end the test manually, then again the issue comes up after a few songs. 
So for me it is more or less clear that this has nothing to do with router settings or the network provider. 
I would also say that it is unlikely that there is a hardware issue with HomePod as it also happens on AirPort Express. 
Linkplay integration I have not in use, however I more than once thought that changing some certain surrounding aspects solved the issue - so far it always came back.

**Comment by JonathanKang** [comment]
> update: With linkplay integration disabled, this issue is gone.

This still happens even with linkplay integration disabled. I can re-produce this issue with a specific song all the time.

**Comment by JonathanKang** [comment]
Reverting back to version 2.2.2 resolved this issue so far for me. I assume this was introduced in version 2.2.3.

**Comment by JonathanKang** [comment]
Airplay verbose logs in version 2.2.2 when playing the song that causes this issue.
```
2024-11-18 14:51:21.878 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] starting ffmpeg with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -ac 2 -channel_layout stereo -ar 44100 -acodec pcm_s16le -f s16le -i - -acodec pcm_s16le -f s16le -ar 44100 -ac 2 -
2024-11-18 14:51:21.883 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Process ffmpeg started with PID 19928
2024-11-18 14:51:21.895 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue 书房音箱 Acton3 Airplay - crossfade: False
2024-11-18 14:51:21.902 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:21.896] main:447 Binding to 192.168.31.40 [enp0s3] with mask 0xffffff00
2024-11-18 14:51:21.902 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:21.896] main:483 Listening for commands on named pipe /tmp/raop-6611
2024-11-18 14:51:21.902 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:21.896] raopcl_create:715 [0x5647d734c020]: using ALAC coding
2024-11-18 14:51:21.902 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:21.897] main:529 Connecting to player: 00226C66ED8C@书房音箱 Acton3._raop._tcp.local. (192.168.31.68:4515)
2024-11-18 14:51:22.000 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:21.999] raopcl_connect:1003 [0x5647d734c020]: local interface 192.168.31.40
2024-11-18 14:51:22.332 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:22.330] exec_request:718 [0x5647d734b3d0]: Body data len 21
2024-11-18 14:51:22.332 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:22.330] raopcl_connect:1087 [0x5647d734c020]: setting volume as part of connect -21.00
2024-11-18 14:51:22.347 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:22.346] main:538 connected to 192.168.31.68 on port 4515, player latency is 1250 ms
2024-11-18 14:51:22.347 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:22.346] main:548 now 1731912682.1490005759, audio starts at NTP 1731912682.527890135 (in 1025 ms)
2024-11-18 14:51:22.347 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:22.346] raopcl_start_at:374 [0x5647d734c020]: set start time 1731912682.527890135 (ts:76377349281619)
2024-11-18 14:51:22.362 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:22.361] raopcl_accept_frames:417 [0x5647d734c020]: begining to stream hts:0 n:1731912682.1551105964
2024-11-18 14:51:22.363 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:22.361] raopcl_accept_frames:425 [0x5647d734c020]: restarting w/o pause n:1731912682.1551105964, hts:76377349281619
2024-11-18 14:51:22.458 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Input #0, s16le, from 'fd:':
2024-11-18 14:51:22.461 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Duration: N/A, bitrate: 1411 kb/s
2024-11-18 14:51:22.461 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Stream #0:0: Audio: pcm_s16le, 44100 Hz, stereo, s16, 1411 kb/s
2024-11-18 14:51:22.463 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Stream mapping:
2024-11-18 14:51:22.464 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Stream #0:0 -> #0:0 (pcm_s16le (native) -> pcm_s16le (native))
2024-11-18 14:51:22.465 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Output #0, s16le, to 'pipe:':
2024-11-18 14:51:22.467 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Metadata:
2024-11-18 14:51:22.467 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] encoder         : Lavf60.3.100
2024-11-18 14:51:22.467 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Stream #0:0: Audio: pcm_s16le, 44100 Hz, stereo, s16, 1411 kb/s
2024-11-18 14:51:22.467 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Metadata:
2024-11-18 14:51:22.467 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] encoder         : Lavc60.3.100 pcm_s16le
2024-11-18 14:51:24.348 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:24.348] main:580 elapsed milliseconds: 961
2024-11-18 14:51:25.349 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:25.348] main:580 elapsed milliseconds: 1959
2024-11-18 14:51:26.350 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.349] main:580 elapsed milliseconds: 2965
2024-11-18 14:51:26.350 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command TITLE=Raffi - Baa Baa Black Sheep
ARTIST=Raffi
ALBUM=Singable Songs for the Very Young
DURATION=59
PROGRESS=0
ACTION=SENDMETA
2024-11-18 14:51:26.355 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command ARTWORK=http://192.168.31.40:8097/imageproxy?path=%25E3%2580%2590%25E7%25A3%25A8%25E8%2580%25B3%25E6%259C%25B5%25E8%258B%25B1%25E6%2596%2587%25E5%2584%25BF%25E6%25AD%258C%25E6%258E%25A8%25E8%258D%2590%25E3%2580%2591%25E5%25AE%259D%25E5%25AE%259D%25E8%258B%25B1%25E8%25AF%25AD%25E5%2590%25AF%25E8%2592%2599%25E5%25BF%2585%25E5%25A4%2587/Raffi%2520-%2520Baa%2520Baa%2520Black%2520Sheep.flac&provider=filesystem_local--SbYy4bqo&size=500&fmt=jpeg
2024-11-18 14:51:26.356 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.354] CmdPipeReaderThread:175 Received line on named pipe TITLE=Raffi - Baa Baa Black Sheep
2024-11-18 14:51:26.357 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.354] CmdPipeReaderThread:175 Received line on named pipe ARTIST=Raffi
2024-11-18 14:51:26.357 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.354] CmdPipeReaderThread:175 Received line on named pipe ALBUM=Singable Songs for the Very Young
2024-11-18 14:51:26.357 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.354] CmdPipeReaderThread:175 Received line on named pipe DURATION=59
2024-11-18 14:51:26.358 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.354] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=0
2024-11-18 14:51:26.374 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.370] CmdPipeReaderThread:175 Received line on named pipe ACTION=SENDMETA
2024-11-18 14:51:26.374 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.370] CmdPipeReaderThread:282 Sending metadata: (nil)
2024-11-18 14:51:26.390 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.388] CmdPipeReaderThread:175 Received line on named pipe ARTWORK=http://192.168.31.40:8097/imageproxy?path=%25E3%2580%2590%25E7%25A3%25A8%25E8%2580%25B3%25E6%259C%25B5%25E8%258B%25B1%25E6%2596%2587%25E5%2584%25BF%25E6%25AD%258C%25E6%258E%25A8%25E8%258D%2590%25E3%2580%2591%25E5%25AE%259D%25E5%25AE%259D%25E8%258B%25B1%25E8%25AF%25AD%25E5%2590%25AF%25E8%2592%2599%25E5%25BF%2585%25E5%25A4%2587/Raffi%2520-%2520Baa%2520Baa%2520Black%2520Sheep.flac&provider=filesystem_local--SbYy4bqo&size=500&fmt=jpeg
2024-11-18 14:51:26.390 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.388] CmdPipeReaderThread:209 Downloading artwork from URL: http://192.168.31.40:8097/imageproxy?path=%25E3%2580%2590%25E7%25A3%25A8%25E8%2580%25B3%25E6%259C%25B5%25E8%258B%25B1%25E6%2596%2587%25E5%2584%25BF%25E6%25AD%258C%25E6%258E%25A8%25E8%258D%2590%25E3%2580%2591%25E5%25AE%259D%25E5%25AE%259D%25E8%258B%25B1%25E8%25AF%25AD%25E5%2590%25AF%25E8%2592%2599%25E5%25BF%2585%25E5%25A4%2587/Raffi%2520-%2520Baa%2520Baa%2520Black%2520Sheep.flac&provider=filesystem_local--SbYy4bqo&size=500&fmt=jpeg
2024-11-18 14:51:26.764 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:26.764] CmdPipeReaderThread:215 Sending artwork to player...
2024-11-18 14:51:27.350 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:27.349] main:580 elapsed milliseconds: 3962
2024-11-18 14:51:28.350 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:28.349] main:580 elapsed milliseconds: 4968
2024-11-18 14:51:29.351 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:29.351] main:580 elapsed milliseconds: 5966
2024-11-18 14:51:30.352 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:30.351] main:580 elapsed milliseconds: 6964
2024-11-18 14:51:31.353 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:31.352] main:580 elapsed milliseconds: 7969
2024-11-18 14:51:31.353 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=7
2024-11-18 14:51:31.502 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:31.500] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=7
2024-11-18 14:51:32.353 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:32.353] main:580 elapsed milliseconds: 8967
2024-11-18 14:51:33.355 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:33.354] main:580 elapsed milliseconds: 9965
2024-11-18 14:51:34.355 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:34.354] main:580 elapsed milliseconds: 10970
2024-11-18 14:51:35.355 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:35.355] main:580 elapsed milliseconds: 11968
2024-11-18 14:51:36.356 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:36.355] main:580 elapsed milliseconds: 12966
2024-11-18 14:51:36.357 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=12
2024-11-18 14:51:36.520 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:36.519] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=12
2024-11-18 14:51:37.357 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:37.357] main:580 elapsed milliseconds: 13972
2024-11-18 14:51:38.358 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:38.357] main:580 elapsed milliseconds: 14969
2024-11-18 14:51:39.359 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:39.358] main:580 elapsed milliseconds: 15975
2024-11-18 14:51:40.359 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:40.358] main:580 elapsed milliseconds: 16973
2024-11-18 14:51:41.360 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:41.359] main:580 elapsed milliseconds: 17971
2024-11-18 14:51:41.361 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=17
2024-11-18 14:51:41.538 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:41.537] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=17
2024-11-18 14:51:42.360 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:42.360] main:580 elapsed milliseconds: 18976
2024-11-18 14:51:43.361 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:43.360] main:580 elapsed milliseconds: 19974
2024-11-18 14:51:44.361 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:44.361] main:580 elapsed milliseconds: 20980
2024-11-18 14:51:45.362 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:45.361] main:580 elapsed milliseconds: 21977
2024-11-18 14:51:46.362 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:46.361] main:580 elapsed milliseconds: 22975
2024-11-18 14:51:46.363 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=22
2024-11-18 14:51:46.555 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:46.554] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=22
2024-11-18 14:51:47.363 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:47.362] main:580 elapsed milliseconds: 23973
2024-11-18 14:51:48.363 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:48.363] main:580 elapsed milliseconds: 24979
2024-11-18 14:51:49.364 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:49.363] main:580 elapsed milliseconds: 25976
2024-11-18 14:51:50.365 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:50.364] main:580 elapsed milliseconds: 26982
2024-11-18 14:51:51.366 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:51.365] main:580 elapsed milliseconds: 27980
2024-11-18 14:51:51.366 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=27
2024-11-18 14:51:51.575 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:51.574] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=27
2024-11-18 14:51:52.367 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:52.366] main:580 elapsed milliseconds: 28978
2024-11-18 14:51:53.367 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:53.366] main:580 elapsed milliseconds: 29983
2024-11-18 14:51:54.368 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:54.367] main:580 elapsed milliseconds: 30981
2024-11-18 14:51:55.368 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:55.367] main:580 elapsed milliseconds: 31979
2024-11-18 14:51:56.369 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:56.368] main:580 elapsed milliseconds: 32984
2024-11-18 14:51:56.369 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=32
2024-11-18 14:51:56.597 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:56.595] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=32
2024-11-18 14:51:57.369 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:57.369] main:580 elapsed milliseconds: 33982
2024-11-18 14:51:58.369 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:58.369] main:580 elapsed milliseconds: 34980
2024-11-18 14:51:58.761 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:58.761] raopcl_send_chunk:589 [0x5647d734c020]: check n:1040898472 p:1040899713 ts:76377350897299 sn:22357
2024-11-18 14:51:58.761 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] retr: 0, avail: 0, send: 0, select: 0)
2024-11-18 14:51:59.370 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:51:59.370] main:580 elapsed milliseconds: 35986
2024-11-18 14:52:00.371 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:00.370] main:580 elapsed milliseconds: 36983
2024-11-18 14:52:01.371 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:01.370] main:580 elapsed milliseconds: 37989
2024-11-18 14:52:01.372 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=37
2024-11-18 14:52:01.616 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:01.616] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=37
2024-11-18 14:52:02.372 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:02.371] main:580 elapsed milliseconds: 38987
2024-11-18 14:52:03.373 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:03.372] main:580 elapsed milliseconds: 39985
2024-11-18 14:52:04.373 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:04.372] main:580 elapsed milliseconds: 40990
2024-11-18 14:52:05.373 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:05.372] main:580 elapsed milliseconds: 41988
2024-11-18 14:52:06.374 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:06.373] main:580 elapsed milliseconds: 42986
2024-11-18 14:52:06.374 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=42
2024-11-18 14:52:06.388 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:06.387] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=42
2024-11-18 14:52:07.374 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:07.373] main:580 elapsed milliseconds: 43991
2024-11-18 14:52:08.373 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:08.373] main:580 elapsed milliseconds: 44989
2024-11-18 14:52:09.374 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:09.374] main:580 elapsed milliseconds: 45987
2024-11-18 14:52:10.375 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:10.374] main:580 elapsed milliseconds: 46993
2024-11-18 14:52:11.375 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:11.374] main:580 elapsed milliseconds: 47990
2024-11-18 14:52:11.375 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=47
2024-11-18 14:52:11.408 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:11.407] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=47
2024-11-18 14:52:12.375 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:12.375] main:580 elapsed milliseconds: 48988
2024-11-18 14:52:13.376 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:13.375] main:580 elapsed milliseconds: 49994
2024-11-18 14:52:14.376 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:14.376] main:580 elapsed milliseconds: 50991
2024-11-18 14:52:14.893 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue 书房音箱 Acton3 Airplay
2024-11-18 14:52:15.377 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:15.376] main:580 elapsed milliseconds: 51989
2024-11-18 14:52:16.378 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:16.377] main:580 elapsed milliseconds: 52995
2024-11-18 14:52:16.379 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=52
2024-11-18 14:52:16.427 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:16.426] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=52
2024-11-18 14:52:17.378 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:17.377] main:580 elapsed milliseconds: 53993
2024-11-18 14:52:18.378 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:18.377] main:580 elapsed milliseconds: 54990
2024-11-18 14:52:19.378 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:19.378] main:580 elapsed milliseconds: 55996
2024-11-18 14:52:20.379 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:20.378] main:580 elapsed milliseconds: 56994
2024-11-18 14:52:21.302 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] size=   10253kB time=00:00:59.51 bitrate=1411.4kbits/s speed=1.01x
2024-11-18 14:52:21.302 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] video:0kB audio:10253kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.000000%
2024-11-18 14:52:21.379 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:21.378] main:580 elapsed milliseconds: 57992
2024-11-18 14:52:21.379 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=57
2024-11-18 14:52:21.446 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:21.445] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=57
2024-11-18 14:52:22.378 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:22.378] main:580 elapsed milliseconds: 58269
2024-11-18 14:52:22.901 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:22.901] main:604 end of stream reached
2024-11-18 14:52:23.559 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:52:23.558] main:613 exiting...
```

Airplay verbose logs in version 2.2.3 when playing the song that causes this issue.
```
2024-11-18 14:57:31.588 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] starting ffmpeg with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -ac 2 -channel_layout stereo -ar 44100 -acodec pcm_s16le -f s16le -i - -acodec pcm_s16le -f s16le -ar 44100 -ac 2 -
2024-11-18 14:57:31.591 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Process ffmpeg started with PID 8357
2024-11-18 14:57:31.598 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue 书房音箱 Acton3 Airplay - crossfade: False
2024-11-18 14:57:31.611 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:31.610] main:447 Binding to 192.168.31.40 [enp0s3] with mask 0xffffff00
2024-11-18 14:57:31.612 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:31.611] main:483 Listening for commands on named pipe /tmp/raop-5901
2024-11-18 14:57:31.620 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:31.620] raopcl_create:715 [0x7fd34e686ac0]: using ALAC coding
2024-11-18 14:57:31.624 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:31.624] main:529 Connecting to player: 00226C66ED8C@书房音箱 Acton3._raop._tcp.local. (192.168.31.68:4515)
2024-11-18 14:57:31.729 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:31.728] raopcl_connect:1003 [0x7fd34e686ac0]: local interface 192.168.31.40
2024-11-18 14:57:32.056 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:32.055] exec_request:718 [0x7fd34e68c000]: Body data len 21
2024-11-18 14:57:32.056 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:32.056] raopcl_connect:1087 [0x7fd34e686ac0]: setting volume as part of connect -24.00
2024-11-18 14:57:32.080 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:32.078] main:538 connected to 192.168.31.68 on port 4515, player latency is 1250 ms
2024-11-18 14:57:32.080 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:32.078] main:548 now 1731913052.336463443, audio starts at NTP 1731913051.3566416288 (in 1002 ms)
2024-11-18 14:57:32.082 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:32.078] raopcl_start_at:374 [0x7fd34e686ac0]: set start time 1731913051.3566416288 (ts:76377365585719)
2024-11-18 14:57:32.101 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:32.100] raopcl_accept_frames:417 [0x7fd34e686ac0]: begining to stream hts:0 n:1731913052.429861801
2024-11-18 14:57:32.103 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:32.100] raopcl_accept_frames:425 [0x7fd34e686ac0]: restarting w/o pause n:1731913052.429861801, hts:76377365585719
2024-11-18 14:57:32.331 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Input #0, s16le, from 'fd:':
2024-11-18 14:57:32.331 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Duration: N/A, bitrate: 1411 kb/s
2024-11-18 14:57:32.331 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Stream #0:0: Audio: pcm_s16le, 44100 Hz, stereo, s16, 1411 kb/s
2024-11-18 14:57:32.331 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Stream mapping:
2024-11-18 14:57:32.332 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Stream #0:0 -> #0:0 (pcm_s16le (native) -> pcm_s16le (native))
2024-11-18 14:57:32.335 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Output #0, s16le, to 'pipe:':
2024-11-18 14:57:32.338 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Metadata:
2024-11-18 14:57:32.338 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] encoder         : Lavf60.16.100
2024-11-18 14:57:32.338 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Stream #0:0: Audio: pcm_s16le, 44100 Hz, stereo, s16, 1411 kb/s
2024-11-18 14:57:32.339 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] Metadata:
2024-11-18 14:57:32.341 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c.ffmpeg] encoder         : Lavc60.31.102 pcm_s16le
2024-11-18 14:57:34.080 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:34.080] main:580 elapsed milliseconds: 977
2024-11-18 14:57:35.080 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:35.080] main:580 elapsed milliseconds: 1983
2024-11-18 14:57:36.081 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.080] main:580 elapsed milliseconds: 2981
2024-11-18 14:57:36.081 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command TITLE=Raffi - Baa Baa Black Sheep
ARTIST=Raffi
ALBUM=Singable Songs for the Very Young
DURATION=59
PROGRESS=0
ACTION=SENDMETA
2024-11-18 14:57:36.082 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command ARTWORK=http://192.168.31.40:8097/imageproxy?path=%25E3%2580%2590%25E7%25A3%25A8%25E8%2580%25B3%25E6%259C%25B5%25E8%258B%25B1%25E6%2596%2587%25E5%2584%25BF%25E6%25AD%258C%25E6%258E%25A8%25E8%258D%2590%25E3%2580%2591%25E5%25AE%259D%25E5%25AE%259D%25E8%258B%25B1%25E8%25AF%25AD%25E5%2590%25AF%25E8%2592%2599%25E5%25BF%2585%25E5%25A4%2587/Raffi%2520-%2520Baa%2520Baa%2520Black%2520Sheep.flac&provider=filesystem_local--SbYy4bqo&size=500&fmt=jpeg
2024-11-18 14:57:36.083 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.082] CmdPipeReaderThread:175 Received line on named pipe TITLE=Raffi - Baa Baa Black Sheep
2024-11-18 14:57:36.083 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.082] CmdPipeReaderThread:175 Received line on named pipe ARTIST=Raffi
2024-11-18 14:57:36.083 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.082] CmdPipeReaderThread:175 Received line on named pipe ALBUM=Singable Songs for the Very Young
2024-11-18 14:57:36.083 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.082] CmdPipeReaderThread:175 Received line on named pipe DURATION=59
2024-11-18 14:57:36.083 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.082] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=0
2024-11-18 14:57:36.105 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.105] CmdPipeReaderThread:175 Received line on named pipe ACTION=SENDMETA
2024-11-18 14:57:36.105 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.105] CmdPipeReaderThread:282 Sending metadata: 0x7fffffff
2024-11-18 14:57:36.121 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.121] CmdPipeReaderThread:175 Received line on named pipe ARTWORK=http://192.168.31.40:8097/imageproxy?path=%25E3%2580%2590%25E7%25A3%25A8%25E8%2580%25B3%25E6%259C%25B5%25E8%258B%25B1%25E6%2596%2587%25E5%2584%25BF%25E6%25AD%258C%25E6%258E%25A8%25E8%258D%2590%25E3%2580%2591%25E5%25AE%259D%25E5%25AE%259D%25E8%258B%25B1%25E8%25AF%25AD%25E5%2590%25AF%25E8%2592%2599%25E5%25BF%2585%25E5%25A4%2587/Raffi%2520-%2520Baa%2520Baa%2520Black%2520Sheep.flac&provider=filesystem_local--SbYy4bqo&size=500&fmt=jpeg
2024-11-18 14:57:36.121 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.121] CmdPipeReaderThread:209 Downloading artwork from URL: http://192.168.31.40:8097/imageproxy?path=%25E3%2580%2590%25E7%25A3%25A8%25E8%2580%25B3%25E6%259C%25B5%25E8%258B%25B1%25E6%2596%2587%25E5%2584%25BF%25E6%25AD%258C%25E6%258E%25A8%25E8%258D%2590%25E3%2580%2591%25E5%25AE%259D%25E5%25AE%259D%25E8%258B%25B1%25E8%25AF%25AD%25E5%2590%25AF%25E8%2592%2599%25E5%25BF%2585%25E5%25A4%2587/Raffi%2520-%2520Baa%2520Baa%2520Black%2520Sheep.flac&provider=filesystem_local--SbYy4bqo&size=500&fmt=jpeg
2024-11-18 14:57:36.374 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:36.373] CmdPipeReaderThread:215 Sending artwork to player...
2024-11-18 14:57:37.082 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:37.081] main:580 elapsed milliseconds: 3978
2024-11-18 14:57:38.086 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:38.083] main:580 elapsed milliseconds: 4984
2024-11-18 14:57:39.084 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:39.084] main:580 elapsed milliseconds: 5982
2024-11-18 14:57:40.086 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:40.084] main:580 elapsed milliseconds: 6987
2024-11-18 14:57:41.084 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:41.084] main:580 elapsed milliseconds: 7985
2024-11-18 14:57:41.085 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=7
2024-11-18 14:57:41.098 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:41.093] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=7
2024-11-18 14:57:42.085 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:42.085] main:580 elapsed milliseconds: 8983
2024-11-18 14:57:43.087 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:43.085] main:580 elapsed milliseconds: 9981
2024-11-18 14:57:44.086 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:44.086] main:580 elapsed milliseconds: 10986
2024-11-18 14:57:45.087 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:45.086] main:580 elapsed milliseconds: 11984
2024-11-18 14:57:46.087 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:46.086] main:580 elapsed milliseconds: 12982
2024-11-18 14:57:46.087 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=12
2024-11-18 14:57:46.133 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:46.132] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=12
2024-11-18 14:57:47.087 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:47.087] main:580 elapsed milliseconds: 13988
2024-11-18 14:57:48.088 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:48.087] main:580 elapsed milliseconds: 14985
2024-11-18 14:57:49.088 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:49.088] main:580 elapsed milliseconds: 15983
2024-11-18 14:57:50.089 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:50.088] main:580 elapsed milliseconds: 16981
2024-11-18 14:57:51.089 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:51.089] main:580 elapsed milliseconds: 17986
2024-11-18 14:57:51.090 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=17
2024-11-18 14:57:51.164 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:51.164] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=17
2024-11-18 14:57:52.090 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:52.089] main:580 elapsed milliseconds: 18984
2024-11-18 14:57:53.090 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:53.089] main:580 elapsed milliseconds: 19990
2024-11-18 14:57:54.091 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:54.090] main:580 elapsed milliseconds: 20988
2024-11-18 14:57:55.091 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:55.091] main:580 elapsed milliseconds: 21993
2024-11-18 14:57:56.092 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:56.092] main:580 elapsed milliseconds: 22991
2024-11-18 14:57:56.093 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=22
2024-11-18 14:57:56.200 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:56.200] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=22
2024-11-18 14:57:57.093 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:57.092] main:580 elapsed milliseconds: 23989
2024-11-18 14:57:58.093 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:58.093] main:580 elapsed milliseconds: 24995
2024-11-18 14:57:58.766 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:58.764] raopcl_send_chunk:589 [0x7fd34e686ac0]: check n:1041258476 p:1041259715 ts:76377366773367 sn:3374
2024-11-18 14:57:58.767 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] retr: 10, avail: 0, send: 0, select: 0)
2024-11-18 14:57:59.094 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:57:59.093] main:580 elapsed milliseconds: 25992
2024-11-18 14:58:00.094 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:00.094] main:580 elapsed milliseconds: 26990
2024-11-18 14:58:01.095 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:01.095] main:580 elapsed milliseconds: 27996
2024-11-18 14:58:01.095 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=27
2024-11-18 14:58:01.235 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:01.234] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=27
2024-11-18 14:58:02.095 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:02.095] main:580 elapsed milliseconds: 28993
2024-11-18 14:58:03.095 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:03.095] main:580 elapsed milliseconds: 29991
2024-11-18 14:58:04.095 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:04.095] main:580 elapsed milliseconds: 30997
2024-11-18 14:58:05.095 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:05.095] main:580 elapsed milliseconds: 31995
2024-11-18 14:58:06.096 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:06.096] main:580 elapsed milliseconds: 32992
2024-11-18 14:58:06.097 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=32
2024-11-18 14:58:06.256 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:06.256] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=32
2024-11-18 14:58:07.096 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:07.096] main:580 elapsed milliseconds: 33998
2024-11-18 14:58:08.097 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:08.096] main:580 elapsed milliseconds: 34996
2024-11-18 14:58:09.098 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:09.097] main:580 elapsed milliseconds: 35994
2024-11-18 14:58:10.115 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:10.097] main:580 elapsed milliseconds: 36999
2024-11-18 14:58:11.098 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:11.097] main:580 elapsed milliseconds: 37997
2024-11-18 14:58:11.098 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=37
2024-11-18 14:58:11.277 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:11.276] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=37
2024-11-18 14:58:12.100 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:12.099] main:580 elapsed milliseconds: 38995
2024-11-18 14:58:13.101 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:13.100] main:580 elapsed milliseconds: 40000
2024-11-18 14:58:14.101 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:14.101] main:580 elapsed milliseconds: 40998
2024-11-18 14:58:15.101 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:15.101] main:580 elapsed milliseconds: 41996
2024-11-18 14:58:16.101 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:16.101] main:580 elapsed milliseconds: 43002
2024-11-18 14:58:16.102 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=42
2024-11-18 14:58:16.305 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:16.305] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=42
2024-11-18 14:58:17.102 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:17.101] main:580 elapsed milliseconds: 43999
2024-11-18 14:58:18.105 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:18.102] main:580 elapsed milliseconds: 45005
2024-11-18 14:58:19.104 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:19.103] main:580 elapsed milliseconds: 46003
2024-11-18 14:58:20.104 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:20.103] main:580 elapsed milliseconds: 47001
2024-11-18 14:58:21.104 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:21.104] main:580 elapsed milliseconds: 48006
2024-11-18 14:58:21.105 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] sending command PROGRESS=48
2024-11-18 14:58:21.330 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:21.327] CmdPipeReaderThread:175 Received line on named pipe PROGRESS=48
2024-11-18 14:58:22.105 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:22.105] main:580 elapsed milliseconds: 49004
2024-11-18 14:58:23.106 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:23.105] main:580 elapsed milliseconds: 50002
2024-11-18 14:58:24.106 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:24.106] main:580 elapsed milliseconds: 51007
2024-11-18 14:58:25.108 VERBOSE (MainThread) [music_assistant.airplay.ap00226c66ed8c] [14:58:25.108] main:580 elapsed milliseconds: 52005
```
In the version 2.2.3 logs, there is no the following line which exists in 2.2.2 logs.
> 2024-11-18 14:52:14.893 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue 书房音箱 Acton3 Airplay

Maybe this is why the stream is never ended.

**Comment by philhzss** [comment]
I'm having the same issue

**Comment by marcelveldt** [comment]
> This still happens even with linkplay integration disabled. I can re-produce this issue with a specific song all the time.

You say you can reproduce this issue with a specific song all the time.
Can you share which song that is please ? So far I'm not able to reproduce any of this so I need as much pointers as possible.
Thanks

**Comment by niteguide** [comment]
> > This still happens even with linkplay integration disabled. I can re-produce this issue with a specific song all the time.
> 
> You say you can reproduce this issue with a specific song all the time. Can you share which song that is please ? So far I'm not able to reproduce any of this so I need as much pointers as possible. Thanks

I have spend a considerable amount of time testing (as you can see above) and found no pattern (no specific song, no specific player setting, no specific amount of time). For me it is clear that its not router, network, hardware. Did you really test HomePod Mini repeating 1 Playlist for lets say half a day repeatedly? Sooner or later this came up on every device I tested.
I hope others have some better clues.

**Comment by JonathanKang** [comment]
> > This still happens even with linkplay integration disabled. I can re-produce this issue with a specific song all the time.
> 
> You say you can reproduce this issue with a specific song all the time. Can you share which song that is please ? So far I'm not able to reproduce any of this so I need as much pointers as possible. Thanks

I setup a local development environment using visual studio code and found out that I cannot re-produce this even using that specific song in the dev container.

Is there a way that I can publish my test image(generated in VS Code) as a addon in home assistant? I'd like to test to see which specific commit between 2.2.2 and 2.2.3 causes this issue.

**Comment by marcelveldt** [comment]
> I have spend a considerable amount of time testing (as you can see above) and found no pattern (no specific song, no specific player setting, no specific amount of time). For me it is clear that its not router, network, hardware. Did you really test HomePod Mini repeating 1 Playlist for lets say half a day repeatedly? Sooner or later this came up on every device I tested.

I play music using Airplay all day in my house to a combination of homepod minis, sonos and even a nad amplifier. 
So yeah, I have tested this forever and ever. Nor was I ever able to reproduce this on my testsetup, my dev setup....

There must be some special trigger/condition what causes this.

> I setup a local development environment using visual studio code and found out that I cannot re-produce this even using that specific song in the dev container.

Interesting, which branch did you checkout? Another interesting test could be to run the docker image on your local machine and yet another test would be to run a venv setup on e.g. the terminal addon for HA. Small steps to rule out that maybe the issue lies in the base image (which is based on alpine linux). It could maybe be related to ffmpeg for instance.

> Is there a way that I can publish my test image(generated in VS Code) as a addon in home assistant? I'd like to test to see which specific commit between 2.2.2 and 2.2.3 causes this issue.

I have no idea what you mean with a test image, there is no such thing.
If you want to test a specific branch, you can use the dev add-on for the Music Assistant Server, it allows you to spinup a specific branch or tag

**Comment by brian-maher** [comment]
I can replicate this with Spotify song 5cY8y2XgOfkAh4kSWLFKkz - that song will play via the Sonos provider, and "plays" with no audio via Airplay (with no errors logged anywhere - even with debug logging it looks perfectly normal).

**Comment by returnofthem** [comment]
> Interesting, which branch did you checkout? Another interesting test could be to run the docker image on your local machine and yet another test would be to run a venv setup on e.g. the terminal addon for HA. Small steps to rule out that maybe the issue lies in the base image (which is based on alpine linux). It could maybe be related to ffmpeg for instance.


Further to the ffmpeg suggestion @marcelveldt I've just seen this in the HA Core 2024.11.4 [release notes:](https://www.home-assistant.io/blog/2024/11/06/release-202411/#2024113---november-22)

> Fix unexpected stop of media playback via ffmpeg proxy for ESPhome devices (@mib1185 - [#130788](https://github.com/home-assistant/core/pull/130788))

I'm no engineer so have no idea if it's the same thing, but sounded like a coincidence worth mentioning.

**Comment by niteguide** [comment]
> I can replicate this with Spotify song 5cY8y2XgOfkAh4kSWLFKkz - that song will play via the Sonos provider, and "plays" with no audio via Airplay (with no errors logged anywhere - even with debug logging it looks perfectly normal).

Confirm! Interesting find indeed ... that is the first song where I see this happen with regularity ... 
<img width="1072" alt="Bildschirmfoto 2024-11-23 um 17 00 18" src="https://github.com/user-attachments/assets/84fa5504-db2e-4a62-99f9-e6ae69c2acd6">

**Comment by JonathanKang** [comment]
> Interesting, which branch did you checkout? Another interesting test could be to run the docker image on your local machine and yet another test would be to run a venv setup on e.g. the terminal addon for HA. Small steps to rule out that maybe the issue lies in the base image (which is based on alpine linux). It could maybe be related to ffmpeg for instance.
> 

I checked out the last commit before version 2.2.3. I'm trying to figure out which commit caused this issue, as I cannot reproduce it in version 2.2.2.

> I have no idea what you mean with a test image, there is no such thing. If you want to test a specific branch, you can use the dev add-on for the Music Assistant Server, it allows you to spinup a specific branch or tag

I was trying to use dev container in VS Code to build mass from a specific commit to test exactly which commit caused this issue. But I'm not able to reproduce this issue using self-built mass. the dev add-on can only target a branch or tag, but not a specific commit.

**Comment by JonathanKang** [comment]
I did some further testing using the music assistant dev addon and found out that this issue is caused by this [commit](https://github.com/music-assistant/server/commit/ecbc5d75c9966891816c122cbdb91942e5e7749e).

**Comment by OzGav** [comment]
@marcelveldt please see above

**Comment by OzGav** [comment]
We have identified a problem with ffmpeg. An update has been shipped in 2.4.0b14. Please report if this has been fixed

**Comment by OzGav** [comment]
@niteguide we will close this soon assuming fixed

**Comment by niteguide** [comment]
@OzGav Please keep open as I was not able to test this yet. As pointed out in some related discussions the latest beta introduced some new streaming errors so I wanted to wait.

**Comment by niteguide** [comment]
> We are now on beta 15 so try that.

Okay! Unfortunately not even a single tone comes from HomePod Mini now - meaning: song will not start, and also does not appear as playing on the UI. I did set up everything from scratch, restarted HomePod Mini and router. On Chromecast the song at least starts.

**Comment by OzGav** [comment]
You need to keep testing then to work out what is unique in your situation as many people use HomePods without problems.

**Comment by niteguide** [comment]
Here is the latest logfile (all b15): 
[music-assistant (13).log](https://github.com/user-attachments/files/18436813/music-assistant.13.log)
I will continue to try ... but I can say that on stable channel music was a least playing ...

**Comment by OzGav** [comment]
What we notice is that there is a recurring attempt to play to `kueche (Airplay)` sometimes 3s apart. Do you have a rogue automation running or something?

**Comment by marcelveldt** [comment]
There are multiple issues in this report, making it hard to follow but all of them are identified and/or fixed.

1) Stream sometimes gets stuck and doesn't continue. We have finally narrowed this down to a bug in ffmpeg in combination with alpine but only in combination with the dynamic volume normalization on some (short) tracks . We're now running a beta with a switch to a newer base image and this looks very good.

2) JonathanKang mentioning that changing a value from a certain commit fixes it. Well, that is just a very complicated way to bypass the volume normalization. You could also just have turned off volume normalization for the same effect. Anyways, its the same issue as mentioned with 1.

3) Latest report from niteguide that music is not playing at all after switching to the beta. This one is completely isolated from anything else here. I suggest to fully restart the HA host and otherwise wait for beta 16 (which contains one more fix for airplay).

**Comment by niteguide** [comment]
> There are multiple issues in this report, making it hard to follow but all of them are identified and/or fixed.
> 
> 1. Stream sometimes gets stuck and doesn't continue. We have finally narrowed this down to a bug in ffmpeg in combination with alpine but only in combination with the dynamic volume normalization on some (short) tracks . We're now running a beta with a switch to a newer base image and this looks very good.
> 2. JonathanKang mentioning that changing a value from a certain commit fixes it. Well, that is just a very complicated way to bypass the volume normalization. You could also just have turned off volume normalization for the same effect. Anyways, its the same issue as mentioned with 1.
> 3. Latest report from niteguide that music is not playing at all after switching to the beta. This one is completely isolated from anything else here. I suggest to fully restart the HA host and otherwise wait for beta 16 (which contains one more fix for airplay).

Thanks a lot for summarizing this particular issue(s) and generally for your dedicated work on MA and HA! 
I have now recognized that other people seem to have the same problem with b15 (#3465) - so I will just wait until b16 arrives and then check if the original issue is gone.

**Comment by OzGav** [comment]
beta 16 is available

**Comment by niteguide** [comment]
With beta 16 I was again able to start music on HomePod Mini and I could start testing. Playback worked well for a very long time (7+ hours); now I recognized that although on UI the music appears playing just normal (song after song) there is no sound coming from HomePod Mini. So the difference now is that we have not that one song that appears to be playing forever within MA. In parallel chromecast player just keeps on playing. 

[music-assistant (14).log](https://github.com/user-attachments/files/18456943/music-assistant.14.log)

**Comment by OzGav** [comment]
If the original problem is fixed and another issue covers your current situation then please close this.

**Comment by niteguide** [comment]
> If the original problem is fixed and another issue covers your current situation then please close this.

From a user perspective the original problem is not fixed because song output still stops after a given time (b17). I understand that behind this are of course different technical aspects. From a user perspective however the situation got even more difficult  with the latest betas because the UI now gives absolutely no hint whether there is sound output or not - everything appears to be just fine while actually nothing can be heard. 
So do you want me to open up a new issue or is that something you are already working on concerning one existing issue? 

Here are my latest logfiles, I think the problem occurred around 2025-01-19 01:05:18.225: 
[music-assistant (15).log](https://github.com/user-attachments/files/18468436/music-assistant.15.log)

Thanks a lot for your support!

**Comment by OzGav** [comment]
The point of an issue report is to to highlight a problem and provide concrete information in order for it to be replicated by the dev team and resolved. 

It isn’t ok for others to chime in with “I have something similar” nor for reporters to say my original problem is fixed but while I’m here I have something else. 

In this case your say the original problem is resolved. That is playback stopping. Silent playback on a HomePod is covered by #3472 so you should monitor that. If that gets resolved and you still have a problem or you identify your problem is different then open a new issue.
